pkg/controllers/csi/provisioner/install.go

```diff
@@ -56,17 +56,17 @@ func (provisioner *OneAgentProvisioner) getInstaller(ctx context.Context, dk dyn
 			Arch:          arch.Arch,
 			Flavor:        arch.Flavor,
 			Technologies:  []string{"all"},
-			TargetVersion: dk.OneAgent().GetCodeModulesVersion(),
+			TargetVersion: dk.OneAgent().GetCustomCodeModulesVersion(),
```

Comment:
This change is not necessary and may break things, the difference between the 2 calls is one reads from the status, the other is from the spec.

The spec one is "processed" by the version reconciler and adds the correct one in the status.

Reply:
How would one then tackle to avoid this error?

```
"oneagent-url","msg":"failed to download specific OneAgent package","version":"custom-image"
```

The version was set to `custom-image` which, I assume, was taken somehow from the status. Can it be that the status was not ready at this point?

Reply:
reverted in 66eca33a7